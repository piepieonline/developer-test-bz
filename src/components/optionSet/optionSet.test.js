import React from 'react';
import ReactDOM from 'react-dom';
import OptionSet from './optionSet';

const oneOption = [
    {
        "type": "Colors",
        "values": [
            {
                "value": "Black",
                "productImage": "./assets/ath-msr7-black.jpg"
            },
            {
                "value": "Brown",
                "productImage": "./assets/ath-msr7-brown.jpg"
            }
        ]
    }
];

it('renders the expected HTML when empty', () => {
    const div = document.createElement('div');
    ReactDOM.render(<OptionSet />, div);

    expect(div).toMatchSnapshot();

    ReactDOM.unmountComponentAtNode(div);
});

it('renders the expected HTML when filled with one option (Colours)', () => {
    const div = document.createElement('div');
    ReactDOM.render(<OptionSet options={oneOption} />, div);

    expect(div).toMatchSnapshot();

    ReactDOM.unmountComponentAtNode(div);
});