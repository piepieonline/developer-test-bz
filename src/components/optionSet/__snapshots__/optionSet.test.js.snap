// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected HTML when empty 1`] = `<div />`;

exports[`renders the expected HTML when filled with one option (Colours) 1`] = `
<div>
  <div
    class="options"
  >
    <div>
      <h3
        class="option-header"
      >
        Colors
      </h3>
      <select
        class="option-select"
      >
        <option
          value="0"
        >
          Black
        </option>
        <option
          value="1"
        >
          Brown
        </option>
      </select>
    </div>
  </div>
</div>
`;
